APMrover2/mode.cpp

```diff
@@ -145,6 +145,26 @@ void Mode::get_pilot_desired_lateral(float &lateral_out)
     lateral_out = rover.channel_lateral->get_control_in();
 }
 
+// decode pilot's input and return heading_out (in cd) and speed_out (in m/s)
+void Mode::get_pilot_desired_heading_and_speed(float &heading_out, float &speed_out)
+{
+    // get steering and throttle in the -1 to +1 range
+    float desired_steering = constrain_float(rover.channel_steer->get_control_in() / 4500.0f, -1.0, 1.0f);
```

Comment:
Probably worth fussing about division vs multiplication here - it will be called a lot.

Similarly further down.

Reply:
It's counter the speed comment, but we could actually make this be `rover.channel_steer->get_control_in() / rover.channel_steer->get_range()` which while slower allows us to change the range from 4500 in the future and have this function not need to be revisited. (The existence of this helper is very new, and I introduced it to start working away from hard coded assumptions about the ranges on plane, as they are causing us some issues).

Just wanted to mention it, the 4500 range is most definitely a throughly embedded concept, so this isn't a hold up.

Reply:
Thinking about this more, we have norm_input_dz() so I've switched to use this and avoid the division (in the vehicle code, it's probably still dividing lower down).